### ISSUE TITLE
Support factorial RepeatedTasks

### ISSUE THREAD

**Issue by jonrkarr** [SED-ML/sed-ml#100]
RepeatedTask allows iteration over a single range (or more specifically simultaneous iteration over multiple ranges).

With three ranges, the equivalent Python code looks roughly like this:
```python
for val1, val2, val3 in zip(range1, range2, range3):
    # modify model according to val1, val2, val3
    # exec task with modified model
```

Some workflow systems such as GitHub actions support "matrices" (outer cross product) of ranges. The equivalent Python for this is
```python
for val1 in range1:
    for val2 in range2:
        for val3 in range3:
            # modify model according to val1, val2, val3
            # exec task with modified model
```

Essentially, this could be used to described factorial designs of simulation experiments. Of course this could encoded into the existing Range classes, but this would loose the semantic meaning. For the same reason its useful to have repeated task, its useful to have yet more abstraction.

Unfortunately, I can't think of a way to achieve this with just a minor tweak to the existing schema. This requires a distinct mechanism to define products of ranges from the existing `listOfRanges` which captures co-iterations of ranges.

**Comment by matthiaskoenig** [comment]
I agree. This is important. I currently just write these ranges out and loose the underlying factorial design in the process.

**Comment by jonrkarr** [comment]
Related to this issue, I think it should also be possible to iterate over targets. For example, you might want to iterate over a list of genes and for each gene change a parameter corresponding to knocking out that gene. This is not possible with the current SED-ML design. Tomas highlighted an example like this during our discussion this week. This can kind of iteration is common in larger network models such as flux balance and logical models.

**Comment by matthiaskoenig** [comment]
This should be possible by just using xpath to all genes. I.e. via an xpath expression you can easily select all individual genes as a target. We just would need to document this clearly. But as I understand xpath and SED-ML this is currently supported. 
The problem is that this results in an addional implicit layer of iteration on the level of the subtask. It would be much cleaner/more intuitive to have this iteration on the range.

**Comment by jonrkarr** [comment]
I don't see how this is currently possible. Currently a target could be
used to change many genes at once. There is no way to iterate over targets
one at a time.
